pkg/api/login.go

```diff
@@ -94,6 +96,8 @@ func (hs *HTTPServer) LoginView(c *models.ReqContext) {
 				return
 			}
 			middleware.DeleteCookie(c.Resp, "redirect_to", hs.cookieOptionsFromCfg)
+			// remove subpath if it exists
+			redirectTo = strings.Replace(redirectTo, setting.AppSubUrl, "", 1)
```

Comment:
What if the subUrl is just `-` or `grafana` and this occur in the redirectTo path? We could do something to replace it only in the beginning of the string, or only once from left to right perhaps.

Reply:
that is true, there's a chance of replacing a match in the wrong location.  currently it removes 1 occurrence but it is not positionally constrained

Reply:
Ah that was the `, 1` part I understand now. I think it may always be the first position, from left to right, as it worked that way in [this playground test](https://play.golang.org/p/L0Djy9kFa07) i did.

If we can safely assume redirectTo will contain settings.AppSubUrl, then I think this is fine as it is.

Reply:
removing this change, debug/test shows it is not needed.  this code path is not easy to hit either :)